api/app.py

```diff
@@ -81,36 +81,119 @@ def create_lister():
+
+        cur.execute("SELECT max(building_id) as building_id from building;")
+        building_id = cur.fetchone()["building_id"]
+        cur.close()
+        return {"status": True, "building_id": building_id}
+    except Exception as e:
+        return {"status": False, "message": f"Error with inserting: {e}"} 
+
+# to be used for address dropdown for building info auto-populate (join)
+@app.route('/api/building/get_addresses', methods = ["GET"])
+def get_building_addresses():
+    cur = mysql.connection.cursor()
+    cur.execute("SELECT DISTINCT building_id, address FROM building;")
+    rv = cur.fetchall()
+    addresses = {pair["address"]: pair["building_id"] for pair in rv}
+    cur.close()
+    return addresses
+
+@app.route('/api/unit/get', methods = ["GET"])
+def get_units():
+    # expecting to be called /api/unit/get?id={id} (optional id) or just /api/unit/get
+    id = request.args.get("id")
+    cur = mysql.connection.cursor()
+
+    if id: # return one unit
+        cur.execute("SELECT * FROM AvailableUnit WHERE unit_id = %s;", [id])
+        rv = cur.fetchone()
+    else: # return all units
+        cur.execute(f"SELECT * FROM AvailableUnit;")
+        rv = cur.fetchall()
+
+    cur.close()
+    return {"data": rv} # rv is a dictionary if provided id, otherwise a list of dictionaries
+
+@app.route('/api/unit/delete', methods = ["DELETE"])
+def delete_unit():
+    # expecting to be called /api/unit/get?id={id} 
+    id = request.args.get("id")
+    conn = mysql.connection
+    cur = conn.cursor()
+    
+    success = True
+    message = ""
+
+    cur.execute("SELECT * FROM AvailableUnit WHERE unit_id = %s", [id])
+    rv = cur.fetchone()
+    if not rv:
+        success = False
+        message = f"unit_id {id} doesn't exist so it cannot be deleted!"
+    else:
+        try:
+            cur.execute("DELETE FROM AvailableUnit WHERE unit_id = %s;", [id])
+        except Exception as e:
+            success = False
+            message = f"Error with deleting unit id {id}: {e}"
+            print(message)
+    
+    cur.close()
+    conn.commit()
+    return {"status": success, "message": message}
+
+
+
 @app.route('/api/unit/create', methods = ["POST"])
-def create_unit():
+def list_unit():
     # check if user is logged in
     if "loggedin" not in session:
-        return {"success": False}, 401
+        return {"success": False, "message": "Not logged in!"}, 401
 
     conn = mysql.connection
     cur = conn.cursor()
 
     json_data = request.get_json()
-    # address is used in future to find building_id
-    address = json_data["address"]
-    room = json_data["room"]
-    lease_term = json_data["leaseDuration"]
-    beds = json_data["numBeds"]
-    floor = json_data["floor"]
-    image = json_data["selectedFile"]
-    washrooms = json_data["numWashrooms"]
-    rent = json_data["price"]
+
+    building_id = json_data["building_id"]
+    room = json_data["room_num"] if json_data["room_num"] != "" else None
```

Comment:
small design nit: use json_data.get(key, None) so if isn't needed.

Reply:
doesnt work for when the value is empty string